### PR TITLE
Python server features

### PR DESCRIPTION
**Pull request by RickyAvina**
- Added more context, configurable by `LINES_OF_QUERY_CONTEXT`.
-  Added lemmatization/stemming on queries and documents for a more robust search. 
- Add loading icon when waiting on server
- Displayed "No results" when no results are found.

### CODE REVIEW

File: python/gensim_req.py [hunk 1]
```diff
@@ -22,6 +27,7 @@ def processInput(json_in):
     Returns:
     An array of caption strings
     """
+    # print(lemmatizer.lemmatize_sentence("Try lemmitizing this sentence if you can"))
```
Comment (yimingnzhao) [inline python/gensim_req.py:30]: Remove if not needed

File: python/gensim_req.py [hunk 2]
```diff
@@ -22,6 +27,7 @@ def processInput(json_in):
     Returns:
     An array of caption strings
     """
```
Comment (yimingnzhao) [inline python/gensim_req.py:29]: Also mention the input parameter `json_in`?

File: python/lemmatizer.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        poss.append(pos)
+        words.append(word)
+    if keepWordPOS:
+        return words, lemmas, [None if i == '' else i for i in poss]
+    return lemmas
+
+
+if __name__ == '__main__':
+    lemmas = lemmatize_sentence("I am hearing that many doubted me in my past")
+    print(lemmas)
```
Comment (yimingnzhao) [inline python/lemmatizer.py:84]: Is this only used for local testing?

File: src/main/webapp/WEB-INF/appengine-web.xml [hunk 4]
```diff
@@ -2,6 +2,7 @@
 <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
   <threadsafe>false</threadsafe>
   <sessions-enabled>true</sessions-enabled>
+  <module>enrique</module>
```
Comment (yimingnzhao) [inline src/main/webapp/WEB-INF/appengine-web.xml:5]: oh no

File: python/lemmatizer.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+tagger = PerceptronTagger()
+pos_tag = tagger.tag
+
+'''
+Stemming and Lemmatization both generate the root form of the inflected words.
+The difference is that stem might not be an actual word whereas, lemma is an actual language word.
+'''
+
+
+def lemmatize(ambiguous_word, pos=None, neverstem=True,
```
Comment (alexandrewalsh) [inline python/lemmatizer.py:20]: can you provide param descriptions?

File: python/lemmatizer.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        return morphy_tag[penntag[:2]]
+    except:
+        return None if returnNone else ''
+
+
+def word_tokenize(text):
+    return text.split()
+
+
+def lemmatize_sentence(sentence, neverstem=False, keepWordPOS=False,
```
Comment (alexandrewalsh) [inline python/lemmatizer.py:63]: ditto

File: src/main/webapp/WEB-INF/appengine-web.xml (hunk 4 above)
Comment (alexandrewalsh) [reply src/main/webapp/WEB-INF/appengine-web.xml:5]: oh no

File: python/app.yaml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 runtime: python37
 service: python
 
-instance_class: F4
+instance_class: F4_1G
```
Comment (cassiehan) [inline python/app.yaml:4]: add a comment here specifying why it is needed

File: python/static/resources/corpora/wordnet/LICENSE [hunk 8]
```diff
@@ -0,0 +1,31 @@
+WordNet Release 3.0
```
Comment (cassiehan) [inline python/static/resources/corpora/wordnet/LICENSE:1]: Do we still need it or we will upload it the lib as a binary file?

File: python/app.yaml (hunk 7 above)
Comment (RickyAvina) [reply python/app.yaml:4]: With this new lemmatizer model, we are now consuming ~1.2 GB of memory per request, which can only be served by a F4_1G instance.

File: python/main.py [hunk 9]
```diff
@@ -25,23 +25,29 @@ def root():
...
         # This request is currently not being used
-
-        query = 'hello'
         json_in = '{"captions": [{"text": "hello, world"}, {"text": "forget me"}]}'
-        res = query_phrase(query, json_in)
-        print("RES: " + str(res))
+        sentance = "this is a really cool sentance to analyze"
+
+        # query = 'unrelated'
+        while True:
```
Comment (jiashenwang) [inline python/main.py:32]: Could u explain a little why are we using ```while True```?

File: python/static/resources/corpora/wordnet/LICENSE (hunk 8 above)
Comment (RickyAvina) [reply python/static/resources/corpora/wordnet/LICENSE:1]: We just need the `.kv` file generated from this corpus. I will delete unecessary files.

File: python/main.py (hunk 9 above)
Comment (RickyAvina) [reply python/main.py:32]: This `GET` request is just for testing purposes, I run this locally and ask for queries input in an inf loop to test.

File: python/static/resources/corpora/wordnet/LICENSE (hunk 8 above)
Comment (RickyAvina) [reply python/static/resources/corpora/wordnet/LICENSE:1]: No longer needed, all we need are the `.kv` files. I will remove all other files.

File: python/lemmatizer.py (hunk 3 above)
Comment (RickyAvina) [reply python/lemmatizer.py:84]: Yes, I can remove this though.

File: python/static/resources/corpora/wordnet/LICENSE (hunk 8 above)
Comment (RickyAvina) [reply python/static/resources/corpora/wordnet/LICENSE:1]: Yes, this is part of `WordNet` which is used in the Lemmatizer.

File: python/main.py (hunk 9 above)
Comment (RickyAvina) [reply python/main.py:32]: This has been changed, GET request are now in use

**Comment by RickyAvina** [comment]
Closing pull request, `python-datastore` has updated changes that address reviewer comments.

**Pull request closed without merging**